### ISSUE TITLE
AppInventor - error in generating colors

### ISSUE THREAD

**Issue by patryk84a** [ysfchn/DynamicComponents-AI2#34]
I set the button color to transparent in the designer.
A = 00
R = FF
G = FF
B = FF
The schema generator did the color code 255 which turned out black in the application. The correct color is 16777215 (64bit? why not -4278190081?).
Why is "if A = 00 then color = 255" fixed in the script? This is not true ... we should use the calculus all the time.
Maybe:

```
# Color converter for AI2
def BuildColor(code : str):
    val = str(code)[2:]
    if len(val) != 6:
        A = int(str(val)[0:2], 16)
    R = int(str(val)[2:4], 16)
    G = int(str(val)[4:6], 16)
    B = int(str(val)[6:], 16)
    if A == 0:
        return (B + (G + (R + (256 * A)) * 256) * 256)
    else:
        return (B + (G + (R + (256 * A)) * 256) * 256) - 4294967296
```
but it doesn't work when we set e.g.
R = 00
R = 00
G = FF
B = FF
but it doesn't work when we set e.g.
R = 00
R = 00
G = FF
B = FF
So it's better to take the color 16777215 for A == 0
```
# Color converter for AI2
def BuildColor(code : str):
    val = str(code)[2:]
    if len(val) != 6:
        A = int(str(val)[0:2], 16)
    R = int(str(val)[2:4], 16)
    G = int(str(val)[4:6], 16)
    B = int(str(val)[6:], 16)
    if A == 0:
        return 16777215
    else:
        return (B + (G + (R + (256 * A)) * 256) * 256) - 4294967296
```

**Comment by ysfchn** [comment]
Well, `255` was transparent for me in Kodular, looks like both builders use different color values.

**Comment by patryk84a** [comment]
AppInventor is one big quirk :D

**Comment by StormiFire** [comment]
> Well, `255` was transparent for me in Kodular, looks like both builders use different color values. I will test if `16777215` works for Kodular.

That's a Kodular issue, `255` should not be transparent.

@Mika-89 can you take a look into this?

**Comment by StormiFire** [comment]
@ysfchn @patryk84a I have found a fix for the issue and will include it in #35

**Comment by ysfchn** [comment]
Closing as this has fixed with #35

**Comment by StormiFire** [comment]
@ysfchn can you reopen this?
